### PR TITLE
Fix false-positive undefined-variable with Lambda, IfExp, and :=

### PR DESCRIPTION
**Pull request by cdce8p**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

Fix false-positive `undefined-variable` for `y`.

```py
l = lambda x: y if (y := x) else None
```

### CODE REVIEW

File: pylint/checkers/variables.py [hunk 1]
```diff
@@ -1469,7 +1469,11 @@ def _is_variable_violation(
                             nodes.Return,
                         ),
                     )
-                    and isinstance(defstmt.value, nodes.IfExp)
+                    and (
+                        isinstance(defstmt.value, nodes.IfExp)
+                        or isinstance(defstmt.value, nodes.Lambda)
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/variables.py:1474]:
```suggestion
                        isinstance(defstmt.value, (nodes.IfExp, nodes.Lambda)),
```

**Review by Pierre-Sassoulas — APPROVED** [review]
Made a small suggestion otherwise 👍

File: pylint/checkers/variables.py (hunk 1 above)
Comment (DanielNoord) [reply pylint/checkers/variables.py:1474]: Doesn't `and` get precedence over `or`? I think this is checking for:
`IfExp` or (`Lambda` and `IfExp`).

Comment (Pierre-Sassoulas) [reply pylint/checkers/variables.py:1474]: My bad !

**Pull request merged**